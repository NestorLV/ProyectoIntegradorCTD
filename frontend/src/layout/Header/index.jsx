import React, { useState } from 'react';
import logo from "./img/logoWguest.jpg";
import Styles from "./styles.module.css"
import StylesApp from "../../App.module.css"
import StylesLayout from "../styles.module.css"
import MenuBurger from '../../components/MenuBurger/MenuBurger';
import MenuButton from '../../components/MenuBurger/MenuButton';
import { Link } from "react-router-dom";
import lineOrange from "./img/LineOrange.png"
import verticalLine from "./img/VerticalLine.png"

export default function Header({ setLastLocation, setBookingWithoutLogin, setLoading, activeCreate, activeLogin, isLogged, showBurger, setShowBurger, handleClean, handleFavourite }) {
    const [admin, setAdmin] = useState(true);
    const [adminMenu, setAdminMenu] = useState(false);
    const showUserName = (isLogged) ? `${Styles.user} ${Styles.loggedIn}` : Styles.user;
    const hideButtons = (isLogged) ? `${Styles.buttons} ${Styles.user}` : Styles.buttons;
    const administration = isLogged ? Styles.optionsBox : Styles.hideButton;

    /* const baseUrl = "http://localhost:8080/" */

    function handleLogOut() {
        document.querySelectorAll("nav")[1].classList.remove(`${Styles.openNav}`)
        document.querySelectorAll("nav")[1].classList.add(`${Styles.closeNav}`)
        setLoading(true)

        sessionStorage.setItem("log", "false")
        sessionStorage.removeItem("id")
        sessionStorage.removeItem("email")
        sessionStorage.removeItem("token")
        setLastLocation(window.location.pathname)
    }

    function handleFavouriteClick() {
        handleFavourite()
        handleShowNav()
    }

    function handleShow() {
        setShowBurger(true)
    }
    function handleHide() {
        setShowBurger(false)
        setBookingWithoutLogin(false)

    }

    function handleErrorLogin() {
        setBookingWithoutLogin(false)
    }

    function handleShowNav() {

        document.querySelectorAll("nav")[1].classList.remove(`${Styles.closeNav}`)
        document.querySelectorAll("nav")[1].classList.add(`${Styles.openNav}`)

    }
    function handleCloseNav() {

        document.querySelectorAll("nav")[1].classList.remove(`${Styles.openNav}`)
        document.querySelectorAll("nav")[1].classList.add(`${Styles.closeNav}`)

    }

    let handleAdminMenu = () => {
        setAdminMenu(prevState => !prevState);
    }

    return (
        <div className={Styles.containerHeader}>
            <header className={`${Styles.header} ${StylesApp.delimiter}`}>
                <div className={showBurger === true ? `${Styles.headerTop} ${StylesLayout.opacity} ${StylesApp.delimiterChild}` : `${Styles.headerTop} ${StylesApp.delimiterChild}`}>
                    <Link to="/" className={Styles.home} onClick={handleHide}>
                        <div className={Styles.logo} onClick={handleClean}>
                            <img src={logo} alt="logo" />
                            <h3>Un mundo por descubrir</h3>
                        </div>
                    </Link>
                    <div className={hideButtons}>
                        <Link to="/create" >
                            <button className={activeCreate ? Styles.hideButton : null} >
                                Crear Cuenta
                            </button>
                        </Link>
                        <Link to="/login">
                            <button className={activeLogin ? Styles.hideButton : null} onClick={handleErrorLogin} >
                                Iniciar Sesión
                            </button>
                        </Link>
                    </div>

                    <div className={administration}>
                        {admin &&
                            <div className={administration} onClick={handleAdminMenu}>
                                <>
                                    <h4 className={Styles.administracion}>Administración</h4>
                                    <img className={Styles.verticalLine} src={verticalLine} alt="line" />
                                </>
                            </div>
                        }

                        <div className={showUserName} onMouseMove={handleShowNav} >
                            <div>
                                <div className={Styles.logoName}>
                                    <p>{sessionStorage.getItem("iniciales")}</p>
                                </div>
                                <div className={Styles.text}>
                                    <h3 className={Styles.great}>Hola,</h3>
                                    <h3 className={Styles.name}>{sessionStorage.getItem("name")} {sessionStorage.getItem("surname")}</h3>
                                </div>
                                <span className={Styles.arrow}></span>
                            </div>
                        </div>
                    </div>
                    <MenuButton show={showBurger} handleShow={handleShow} />
                </div>
                <MenuBurger show={showBurger} handleHide={handleHide} isLogged={isLogged} iniciales={sessionStorage.getItem("iniciales")} activeLogin={activeLogin} activeCreate={activeCreate} handleLogOut={handleLogOut} handleFavourite={handleFavourite} />
            </header>
            <nav className={Styles.closeNav} onMouseLeave={handleCloseNav}>
                <Link to="/">
                    <h4 className={Styles.seeMyAccount} onClick={handleFavouriteClick}>Ver favoritos</h4>
                </Link>
                <img src={lineOrange} alt="" />
                <Link to="/">
                    <h4 className={Styles.seeMyAccount}>Mis reservas</h4>
                </Link>
                <img src={lineOrange} alt="" />
                <h4 className={Styles.seeMyAccount}><a href="/" onClick={handleLogOut}>Cerrar sesión</a> </h4>
            </nav>
            <div className={adminMenu ? Styles.adminOptions : Styles.hideButton} >
                <Link to="/product/create">
                    <h4 className={Styles.seeMyAccount}>Crear Producto</h4>
                </Link>
                <img src={lineOrange} alt="" />
                <Link to="/product/update">
                    <h4 className={Styles.seeMyAccount}>Modificar Producto</h4>
                </Link>
            </div>
        </div >
    )
}