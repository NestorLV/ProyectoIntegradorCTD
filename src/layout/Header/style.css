header.header {
    align-items: center;
    background-color: white;
    display: flex;
    flex-wrap: nowrap;
    height: 80px;
    justify-content: space-between;
    padding: 5px 32px;
    position: fixed;
    width: 100%;
    z-index: 100;
  }
  
  /* css contenedor logo  */
  header.header a {
    text-decoration: none;
  }
  
  .home div.logo {
    align-items: flex-end;
    display: flex;
  }
  
  .home div.logo img {
    height: 52px;
    width: 71px;
  }
  
  .home div.logo h3 {
    color: var(--tertiary-color);
    display: flex;
    font-size: var(--heading3);
    font-style: italic;
    font-weight: var(--font-regular);
    padding-left: 12px;
    width: 270px;
  }
  
  /* css contenedor botones  */
  
  div.buttons {
    display: flex;
    padding-right: 8px;
  }
  
  div.buttons button {
    background-color: white;
    border-radius: 5px;
    border: 1px solid var(--primary-color);
    color: var(--primary-color);
    font-size: var(--button1);
    font-weight: var(--font-bold);
    margin: 0 10px;
    padding: 5pt;
    width: 206px;
  }
  
  /* css contenedor user  */
  
  div.user {
    display: none; /* cambiar a flex para mostrarlo */
  }
  
  div.user div {
    padding: 0 5px;
  }
  
  div.user div.logoName {
    align-items: center;
    display: flex;
    flex-wrap: wrap;
  }
  
  div.user div.logoName p {
    background-color: var(--secondary-color);
    border-radius: 50%;
    color: white;
    font-size: var(--heading2);
    font-weight: var(--font-bold);
    padding: 5px;
  }
  
  div.user div.text h3 {
    font-size: var(--heading3);
  }
  
  div.user div.text h3.great {
    opacity: 50%;
  }
  
  div.user div.text h3.name {
    color: var(--primary-color);
  }
  
  div.user div.close {
    position: relative;
    right: 20px;
    top: -10px;
  }
  
  /* css contenedor mobile  */
  
  div.mobile {
    display: none; /* cambiar a block para mostrarlo */
    justify-content: center;
    margin-left: 20px;
  }
  
  /* media queries */
  
  @media all and (max-width: 769px) {
    /* Header */
    a.home div.logo h3 {
      display: none;
    }
  
    div.buttons button {
      width: 164px;
    }
  }
  
  @media all and (max-width: 415px) {
    /* Header */
  
    header.header {
      height: 93px;
      padding: 5px 11px;
    }
  
    a.home div.logo img {
      height: 50px;
    }
  
    div.buttons {
      display: none;
    }
  
    div.mobile {
      display: block;
    }
  }