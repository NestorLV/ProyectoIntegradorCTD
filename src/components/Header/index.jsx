import React from 'react';
import './style.css';

export default function Header() {
    return (
        <header className="header">
            <a href="/" className="home">
                <div className="logo">
                    <img src={"img/logo 1DB.png"} alt="logo"/>
                    <h3>Sentite como en tu hogar</h3>
                </div>
            </a>
            <div className="buttons">
                <button>Crear Cuenta</button>
                <button>Iniciar Sesion</button>
            </div>
            <div className="user">
                <div className="logoName">
                    <p>BR</p>
                </div>
                <div className="text">
                    <h3 className="great">Hola,</h3>
                    <h3 className="name">Bruno Rodriguez</h3>
                </div>

                <div className="close"><a href="/">X</a> </div>
            </div>
            <div className="mobile">
                <img src={"img/menÃº.png"} alt="menu mobile"/>
            </div>
        </header>
    )
}