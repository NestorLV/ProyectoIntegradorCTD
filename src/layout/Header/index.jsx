//import { PROPERTY_TYPES } from '@babel/types';
import React, { useState, useEffect } from 'react';
import { Link } from "react-router-3";
import ValidCredentials from '../../credentials/ValidCredentials';
import logo from "./img/logo 1DB.png";
import './style.css';
import MenuBurger from '../../components/MenuBurger/MenuBurger';
import { HeaderWrapper } from '../../components/MenuBurger/HeaderStyles';
import MenuButton from '../../components/MenuBurger/MenuButton';

export default function Header(props) {

    const [activeCreate, setIsActiveCreate] = useState((window.location.pathname==="/create")?true:false)
    const [activeLogin, setIsActiveLogin] = useState((window.location.pathname==="/login")?true:false)
    const [activeHome, setActiveHome] = useState((window.location.pathname==="/")?true:false)

    /*function toggleClass(name) {
        const ruta = window.location.pathname;
        console.log("headerRuta", ruta);
        if (name === "createButton") {
            setIsActiveCreate(true)
            setIsActiveLogin(false)
        } else if (name === "loginButton") {
            setIsActiveLogin(true)
            setIsActiveCreate(false)
        } else if("home"||ruta==="/"){
            setIsActiveLogin(false)
            setIsActiveCreate(false)
        }
    };*/
    const ruta = window.location.pathname;
    console.log("ruta",ruta);

    function hide() {
        if (ruta==="/login") {
            setIsActiveCreate(false)
            setIsActiveLogin(true)
        } else if (ruta==="/create") {
            setIsActiveLogin(false)
            setIsActiveCreate(true)
        } else if(ruta==="/"){
            setIsActiveLogin(false)
            setIsActiveCreate(false)
        }
    };
    
    useEffect(
        ()=> hide()
    )
    
    console.log("header ruta",window.location.pathname);
    console.log("activeLogin", activeLogin);
    console.log("activeCreate", activeCreate);
    const showUserName = (props.isLogged) ? "user loggedIn" : "user"; //para cambiar la clase segun este loggueado o no y
    //que aparezca el nombre del usuario
    const hideButtons = (props.isLogged) ? "buttons user" : "buttons";//Para esconder los botones cuando esta logueado
    const iniciales = `${ValidCredentials.nombre.slice(0, 1).toUpperCase()}${ValidCredentials.apellido.slice(0, 1).toUpperCase()}`

    

    function handleLogOut() {
        sessionStorage.setItem("log", "false")
    }
    
    const [showBurger, setShowBurger] = useState(false)

    function handleShow(){
        setShowBurger(true)
    }
    function handleHide(){
        setShowBurger(false)
    }

    return (
        <HeaderWrapper className="header">
            <Link to="/" className="home" onClick={handleHide}>
                <div className="logo">
                    <img src={logo} alt="logo" />
                    <h3>Sentite como en tu hogar</h3>
                </div>
            </Link>
            <div className={hideButtons}>
                <Link to="/create"><button className={activeCreate ? "hideButton" : null} >Crear Cuenta</button></Link>
                <Link to="/login"><button className={activeLogin ? "hideButton" : null} >Iniciar Sesion</button></Link>
            </div>
            <div className={showUserName}>
                <div className="logoName">
                    <p>{iniciales}</p>
                </div>
                <div className="text">
                    <h3 className="great">Hola,</h3>
                    <h3 className="name">{ValidCredentials.nombre} {ValidCredentials.apellido}</h3>
                </div>
                <div className="close"><a href="/" onClick={handleLogOut}>X</a> </div>
            </div>
            <MenuBurger show={showBurger} handleShow={handleShow} handleHide={handleHide} isLogged={props.isLogged} iniciales={iniciales} activeLogin={activeLogin} activeCreate ={activeCreate} handleLogOut={handleLogOut}/>
            <MenuButton show={showBurger} handleShow={handleShow} /> 
        </HeaderWrapper>
    )
}