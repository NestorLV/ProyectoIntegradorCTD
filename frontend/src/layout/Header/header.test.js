import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Header from './index';

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe("Probando el componente <Header/>", () => {
    let wrapper;
    let setActiveCreate = jest.fn(); //Simula la funcion del set
    let setActiveLogin = jest.fn() //Simula la funcion del set
   
    beforeEach(() => {
        wrapper = shallow(<Header setActiveCreate={setActiveCreate} setActiveLogin={setActiveLogin} />)
    });

    it("Deberia mostrar <Header/> correctamente", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("Deberia retornar 3 etiquetas h3", () => {
        let node = wrapper.find("h3")
        expect(node.length).toEqual(3)
    });

    it("Deberia retornar 3 Links", () => {
        let node = wrapper.find("Link")
        expect(node.length).toEqual(3)
    });

    it("Deberia retornar 2 botones", () => {
        let node = wrapper.find("button")
        expect(node.length).toEqual(2)
    });

});

