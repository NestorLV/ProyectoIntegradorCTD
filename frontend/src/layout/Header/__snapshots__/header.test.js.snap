// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el componente <Header/> Deberia mostrar <Header/> correctamente 1`] = `
<header
  className="header delimiter"
>
  <div
    className="headerTop delimiterChild"
  >
    <Link
      className="home"
      onClick={[Function]}
      to="/"
    >
      <div
        className="logo"
        onClick={[MockFunction]}
      >
        <img
          alt="logo"
          src="logoWguest.jpg"
        />
        <h3>
          Un mundo por descubrir
        </h3>
      </div>
    </Link>
    <div
      className="buttons"
    >
      <Link
        to="/create"
      >
        <button
          className={null}
        >
          Crear Cuenta
        </button>
      </Link>
      <Link
        to="/login"
      >
        <button
          className={null}
          onClick={[Function]}
        >
          Iniciar Sesi√≥n
        </button>
      </Link>
    </div>
    <div
      className="user"
    >
      <div
        className="logoName"
      >
        <p />
      </div>
      <div
        className="text"
      >
        <h3
          className="great"
        >
          Hola,
        </h3>
        <h3
          className="name"
        >
           
        </h3>
        <h4
          className="seeFavourite"
          onClick={[MockFunction]}
        >
          Ver favoritos
        </h4>
      </div>
      <div>
        <div
          className="close"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            X
          </a>
           
        </div>
      </div>
    </div>
    <MenuButton
      handleShow={[Function]}
    />
  </div>
  <MenuMobile
    handleFavourite={[MockFunction]}
    handleHide={[Function]}
    handleLogOut={[Function]}
    iniciales={null}
  />
</header>
`;
